File: week1/tower_of_hanoi.py
def hanoi(n, source, target, auxiliary, moves):
    if n == 1:
        moves.append((source, target))
        return
    hanoi(n-1, source, auxiliary, target, moves)
    moves.append((source, target))
    hanoi(n-1, auxiliary, target, source, moves)

n = int(input())
moves = []
hanoi(n, 1, 3, 2, moves)
print(len(moves))
for move in moves:
    print(move[0], move[1])

File: Chavda_23B0368/Week_1/Tower_of_hanoi.cpp
#include <iostream>
#include <vector>
using namespace std;

void hanoi(int n, int source, int target, int auxiliary, int& count, vector<vector<int>>& moves) {
    if (n == 1) {
        moves.push_back({source, target});
        count++;
        return;
    }
    hanoi(n - 1, source, auxiliary, target, count, moves);
    moves.push_back({source, target});
    count++;
    hanoi(n - 1, auxiliary, target, source, count, moves);
}

int main() {
    int n;
    cin >> n;

    vector<vector<int>> moves;
    int count = 0;

    hanoi(n, 1, 3, 2, count, moves);

    cout << count << "\n";
    for (const auto& move : moves) {
        cout << move[0] << " " << move[1] << "\n";
    }

    return 0;
}

